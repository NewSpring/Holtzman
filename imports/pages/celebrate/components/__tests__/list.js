import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import {
  getRatio,
  dynamicWidthContainer,
  dynamicWidth,
  dynamicSize,
  ListItem,
  ListWithoutData as List,
} from "../list";

describe("getRatio", () => {
  it("returns `0.8` when under `480", () => {
    const result = getRatio(100);
    expect(result).toBe(0.8);
  });

  it("returns `0.6` when under `768`", () => {
    const result = getRatio(700);
    expect(result).toBe(0.6);
  });

  it("returns `0.4` when under `1025`", () => {
    const result = getRatio(1000);
    expect(result).toBe(0.4);
  });

  it("returns `0.2` when larger than `1025`", () => {
    const result = getRatio(2000);
    expect(result).toBe(0.2);
  });
});

it("dynamicWidthContainer uses window to calculate width", () => {
  const result = dynamicWidthContainer(2);
  expect(result).toMatchSnapshot();
});

it("dynamicWidth uses window to calculate width", () => {
  const result = dynamicWidth(2);
  expect(result).toMatchSnapshot();
});

describe("No window", () => {
  beforeEach(() => {
    delete global.window;
  });

  it("dynamicWidthContainer returns blank object if no window", () => {
    const result = dynamicWidthContainer(2);
    expect(result).toEqual({});
  });

  it("dynamicWidth returns blank object if no window", () => {
    const result = dynamicWidth(2);
    expect(result).toEqual({});
  });
});

describe("dynamicSize", () => {
  it("returns `0.2` for less than `4`", () => {
    expect(dynamicSize("aa")).toBe(0.2);
  });
  it("returns `0.35` for less than `6`", () => {
    expect(dynamicSize("aaaaa")).toBe(0.35);
  });
  it("returns `0.4` for less than `8`", () => {
    expect(dynamicSize("aaaaaaa")).toBe(0.4);
  });
  it("returns `0.5` for less than `10`", () => {
    expect(dynamicSize("aaaaaaaaa")).toBe(0.5);
  });
  it("returns `0.6` for less than `12`", () => {
    expect(dynamicSize("aaaaaaaaaaa")).toBe(0.6);
  });
  it("returns `0.7` for `12` or more", () => {
    expect(dynamicSize("aaaaaaaaaaaa")).toBe(0.7);
  });
});

describe("ListItem", () => {
  const defaultProps = {
    item: {
      count: 2,
      label: "test",
    },
    padding: false,
  };

  const generateComponent = (additionalProps = {}) => {
    const newProps = {
      ...defaultProps,
      ...additionalProps,
    };
    return <ListItem { ...newProps } />;
  };

  it("renders with props", () => {
    const wrapper = shallow(generateComponent());
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it("adds push-right if padding", () => {
    const wrapper = shallow(generateComponent({
      padding: true,
    }));
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe("List", () => {
  const defaultProps = {
    items: [
      {
        count: 2,
        label: "test",
      },
      {
        count: 2,
        label: "test",
      },
    ],
    width: 700,
  };

  const generateComponent = () => (
    <List { ...defaultProps } />
  );

  it("renders with props", () => {
    const wrapper = shallow(generateComponent());
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
