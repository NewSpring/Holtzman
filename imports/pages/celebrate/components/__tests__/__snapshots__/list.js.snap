exports[`List renders with props 1`] = `
<div
  style={
    Object {
      "WebkitOverflowScrolling": "touch",
      "overflowX": "scroll",
      "overflowY": "hidden",
    }
  }>
  <section
    style={Object {}}>
    <ListItem
      item={
        Object {
          "count": 2,
          "label": "test",
        }
      }
      padding={true} />
    <ListItem
      item={
        Object {
          "count": 2,
          "label": "test",
        }
      }
      padding={false} />
  </section>
</div>
`;

exports[`ListItem adds push-right if padding 1`] = `
<div
  className="card floating display-inline-block push-right"
  style={Object {}}>
  <div
    className="floating__item one-whole soft">
    <Connect(FitTextWithoutData)
      compressor={0.7}>
      <h1
        className="uppercase flush-bottom soft-half-bottom"
        style={
          Object {
            "fontWeight": "900",
          }
        }>
        2
      </h1>
    </Connect(FitTextWithoutData)>
    <Connect(FitTextWithoutData)
      compressor={1.75}>
      <h5
        className="flush-bottom">
        test
      </h5>
    </Connect(FitTextWithoutData)>
  </div>
</div>
`;

exports[`ListItem renders with props 1`] = `
<div
  className="card floating display-inline-block "
  style={Object {}}>
  <div
    className="floating__item one-whole soft">
    <Connect(FitTextWithoutData)
      compressor={0.7}>
      <h1
        className="uppercase flush-bottom soft-half-bottom"
        style={
          Object {
            "fontWeight": "900",
          }
        }>
        2
      </h1>
    </Connect(FitTextWithoutData)>
    <Connect(FitTextWithoutData)
      compressor={1.75}>
      <h5
        className="flush-bottom">
        test
      </h5>
    </Connect(FitTextWithoutData)>
  </div>
</div>
`;

exports[`test dynamicWidth uses window to calculate width 1`] = `
Object {
  "height": 393.6,
  "width": 393.6,
}
`;

exports[`test dynamicWidthContainer uses window to calculate width 1`] = `
Object {
  "width": "867.2px",
}
`;
